src/createSettingsFileTransformer.js

```diff
@@ -0,0 +1,140 @@
+  // nothing to do
+  if (!pathSegments.length || !isPlainObject(settings)) {
+    return;
+  }
+
+  var currentKey = pathSegments[0];
+
+  // base case
+  if (pathSegments.length === 1) {
+    if (
+      settings.hasOwnProperty(currentKey) &&
+      typeof settings[currentKey] === 'string'
+    ) {
+      settings[currentKey] = decorateWithDynamicHost(settings[currentKey]);
+    }
+    return;
+  }
+
+  // still more work to do
+  var remainingPathSegments = pathSegments.slice(1);
+  if (isArrayReference(currentKey)) {
+    // 'someArrayReference[]' --> 'someArrayReference'
+    currentKey = sanitizeArrayKey(currentKey);
+    var settingsValue = settings[currentKey];
+    if (Array.isArray(settingsValue)) {
+      settingsValue.forEach(function (arrayEntryObject) {
+        return traverseIntoSettings(
+          remainingPathSegments,
+          arrayEntryObject,
+          decorateWithDynamicHost
+        );
+      });
+    }
+  } else {
+    // object case
+    return traverseIntoSettings(
+      remainingPathSegments,
+      settings[currentKey],
+      decorateWithDynamicHost
+    );
+  }
+}
+
+/**
+ * Returns a function that when called will attempt to traverse the passed in
+ * settings object to each file path and transform a relative URL to an absolute
+ * URL.
+ *
+ * @param isDynamicEnforced
+ * @param decorateWithDynamicHost
+ * @returns {(function(*=, *=, *=): (*))|*}
+ */
+module.exports = function (isDynamicEnforced, decorateWithDynamicHost) {
+  return function (settings, filePaths, moduleReferencePath) {
+    if (
+      !isDynamicEnforced ||
+      !isPlainObject(settings) ||
+      !Object.keys(settings).length ||
+      !Array.isArray(filePaths) ||
+      !filePaths.length
+    ) {
+      return settings;
+    }
+
+    // pull out the file paths by the module's reference path and loop over each urlPath
+    filePaths.forEach(function (filePathString) {
+      // The custom code action provides the ability to have the source code in the 'source'
+      // variable or to have an external file. Therefore, this module has 2 behaviors.
+      // It also does not provide a value of false for isExternal just as all other extensions
+      // that use fileTransform do not provide an isExternal variable check. Therefore, we need
+      // to treat Adobe's custom code action special, and don't augment the 'source' variable
+      // if isExternal is not also present.
+      var isAdobeCustomCodeAction = Boolean(
+        moduleReferencePath != null &&
+          /^core\/.*actions.*\/customCode\.js$/.test(moduleReferencePath)
+      );
+      if (
+        isAdobeCustomCodeAction &&
+        filePathString === 'source' &&
+        !settings.isExternal
+      ) {
+        return;
+      }
+
+      try {
+        // modify the object in place
+        traverseIntoSettings(
+          filePathString.split('.'),
+          settings,
+          decorateWithDynamicHost
+        );
+      } catch (e) {
```

Comment:
Since we are not longer throw an error inside traverseIntoSettings, you can remove this try-catch statement.